sources/packages/api/src/services/sequence-control/sequence-control.service.ts

```diff
@@ -50,14 +50,51 @@ export class SequenceControlService extends RecordDataModelService<SequenceContr
     );
     try {
       await queryRunner.startTransaction();
+      await this.consumeNextSequenceWithExistingEntityManager(
+        sequenceName,
+        process,
+        queryRunner.manager,
+      );
+      await queryRunner.commitTransaction();
+    } catch (error) {
+      this.logger.error("Executing sequence number rollback...");
+      await queryRunner.rollbackTransaction();
+      throw error;
+    } finally {
+      await queryRunner.release();
+    }
+  }
+
+  /**
+   * Allows that a new sequence number, controlled by the database, be acquired.
+   * Between the process of getting the next sequence number and updating it back
+   * to the database, a process can be executed in the middle, making it possible that
+   * the number on database only be incremented if the process in the middle
+   * was successfully  executed.
+   * @param sequenceName name of the sequence to be incremented.
+   * @param process process to be executed. Depending on the process
+   * to be executed, in case of an error happen, the sequence could not be
+   * consumed in case of failure.
+   * @param entityManager existing entity manager, to use the transaction from the
+   * called method.
+   */
+  public async consumeNextSequenceWithExistingEntityManager(
+    sequenceName: string,
+    process: (
+      sequenceNumber: number,
+      entityManager: EntityManager,
+    ) => Promise<void>,
+    entityManager: EntityManager,
```

Comment:
Can we move this parameter to the second position? I believe that it would be cleaner when it is being invoked. Does it make sense?
The _process_ parameter could be big enough to make the _entityManager_ be lost a bit at the end of the list of parameters when calling it.